CMakeLists.txt

```diff
@@ -95,3 +95,17 @@ set_target_properties(
     CXX_STANDARD_REQUIRED true
     XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "Developer ID Application: Daniel Kroening"
 )
+
+find_package(Git)
+if(GIT_FOUND)
+   message(STATUS "Found Git executable: ${GIT_EXECUTABLE}")
+   execute_process(
+       COMMAND "${GIT_EXECUTABLE}" "describe" "--tags" "--always" "--long"
+       OUTPUT_VARIABLE git_repo_version
+       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+       OUTPUT_STRIP_TRAILING_WHITESPACE
+   )
+   message(STATUS "Executable is version ${git_repo_version}")
+else()
+   message(FATAL_ERROR "Git is not installed, cannot generate version string")
```

Comment:
won't this stop the build in case of no git ? If I understand correctly the make build would use `n/a` as version string in that case.

Reply:
I might have done too much of a copy&paste :-)  I'll take another look.